Xamarin.Essentials/Permissions/Permissions.ios.tvos.watchos.cs

```diff
@@ -269,5 +270,74 @@ public partial class StorageWrite : BasePlatformPermission
         public partial class Vibrate : BasePlatformPermission
         {
         }
+
+        internal static class PushNotificationPermissions
+        {
+            public static async Task<PermissionStatus> RequestAsync(UNAuthorizationOptions authorizationOptions)
+            {
+                var (_, error) = await UNUserNotificationCenter.Current.RequestAuthorizationAsync(authorizationOptions);
+                if (error != null)
+                {
+                    throw new NSErrorException(error);
+                }
+
+                return await CheckStatusAsync();
+            }
+
+            public static async Task<PermissionStatus> CheckStatusAsync()
+            {
+                var settings = await UNUserNotificationCenter.Current.GetNotificationSettingsAsync();
+                var authorizationStatus = settings.AuthorizationStatus;
+                return authorizationStatus switch
+                {
+                    UNAuthorizationStatus.NotDetermined => PermissionStatus.Unknown,
+                    UNAuthorizationStatus.Denied => PermissionStatus.Denied,
+                    UNAuthorizationStatus.Authorized => PermissionStatus.Granted,
+                    UNAuthorizationStatus.Provisional => PermissionStatus.Restricted,
+                    UNAuthorizationStatus.Ephemeral => PermissionStatus.Granted,
+                    _ => throw new ArgumentOutOfRangeException(
+                        paramName: nameof(UNNotificationSettings.AuthorizationStatus),
+                        actualValue: authorizationStatus,
+                        message: null),
+                };
+            }
+        }
+
+        public partial class PushNotification : BasePlatformPermission
+        {
+            public override Task<PermissionStatus> RequestAsync()
+            {
+                EnsureDeclared();
```

Comment:
There is no `RequiredInfoPlistKeys` override. I assume that we want to check if there is a plist entry/entries for push notifications?